sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -1152,30 +1200,37 @@ void closeFileSystem(String fileSystemName) {
         this.openFileSystems.remove(fileSystemName);
     }
 
-    private String extractAccountEndpoint(URI uri) {
+    private String extractAccountEndpointOrGetDefault(URI uri) {
         if (!uri.getScheme().equals(this.getScheme())) {
             throw LoggingUtility.logError(ClientLoggerHolder.LOGGER, new IllegalArgumentException(
                 "URI scheme does not match this provider"));
         }
-        if (CoreUtils.isNullOrEmpty(uri.getQuery())) {
-            throw LoggingUtility.logError(ClientLoggerHolder.LOGGER,
-                new IllegalArgumentException("URI does not contain a query component. FileSystems require a URI of "
-                    + "the format \"azb://?endpoint=<account_endpoint>\"."));
-        }
 
-        String endpoint = Flux.fromArray(uri.getQuery().split("&"))
-                .filter(s -> s.startsWith(ENDPOINT_QUERY_KEY + "="))
-                .switchIfEmpty(Mono.defer(() -> Mono.error(LoggingUtility.logError(ClientLoggerHolder.LOGGER,
-                    new IllegalArgumentException("URI does not contain an \"" + ENDPOINT_QUERY_KEY + "=\" parameter. "
-                        + "FileSystems require a URI of the format \"azb://?endpoint=<endpoint>\"")))))
-                .map(s -> s.substring(ENDPOINT_QUERY_KEY.length() + 1)) // Trim the query key and =
-                .blockLast();
+        String endpoint = Arrays.stream((uri.getQuery() != null ? uri.getQuery() : "").split("&"))
```

Comment:
I'd split this into if has query do splitting otherwise just use the environment endpoint, no need to do all the extra processing for a known case. May want to check `!CoreUtils.isNullOrEmpty(uri.getQuery())`